src/JSTest.Runner/ObjectModel/Common/TestCase.ts

```diff
@@ -30,5 +36,51 @@ export class TestCase {
             .appendStr(this.Source);
 
         this.Id = hash.getGuid();
+
+        this.AttachmentGuid = null;
+        if (attachmentId) {
+            const attachmentHash = new Md5();
+            attachmentHash.appendStr(attachmentId);
+            this.AttachmentGuid = attachmentHash.getGuid();
+        }
+    }
+
+    /**
+     * Looks for files to upload under the specified folder using test case specific id.
+     * 
+     * @param attachmentsRootFolder Root folder of the attachments.
+     */
+    public getAttachments(attachmentsRootFolder: string): AttachmentSet[] {
+        const attachments = new Array<AttachmentSet>();
+
+        // Lets see any file exists in the attachments folder upload
+        if (attachmentsRootFolder && this.AttachmentGuid) {
+            const attachmentsFolder = path.join(attachmentsRootFolder, this.AttachmentGuid);
+            if (fs.existsSync(attachmentsFolder) && fs.lstatSync(attachmentsFolder).isDirectory()) {
+                let attachmentSet: AttachmentSet = null;
+
+                // Iterate through the files under attachments folder to get the list of attachments
+                fs.readdirSync(attachmentsFolder).forEach(file => {
+                    const filePath = path.join(attachmentsFolder, file);
+                    const fileStats = fs.lstatSync(filePath);
+                    if (fileStats.isFile()) {
+                        EqtTrace.info(`ExecutionManager: adding set ${this.ExecutorUri}`);
```

Comment:
`TestCase.getAttachments: adding set ${this.ExecutorUri}`

keep an eqtTrace statement when calling this in execution manager, this describes the flow in a better fashion